analyzer/src/main/kotlin/AnalyzerResultBuilder.kt

```diff
@@ -40,7 +41,14 @@ class AnalyzerResultBuilder(private val curationProvider: PackageCurationProvide
     private val issues = sortedMapOf<Identifier, List<OrtIssue>>()
     private val dependencyGraphs = sortedMapOf<String, DependencyGraph>()
 
-    fun build() = DependencyGraphConverter.convert(AnalyzerResult(projects, packages, issues, dependencyGraphs))
+    fun build(): AnalyzerResult {
+        val duplicateIds = (projects.map { it.id } + packages.map { it.pkg.id }).getDuplicates()
+        require(duplicateIds.isEmpty()) {
+            "AnalyzerResult contains packages that are also projects. Duplicates: '$duplicateIds'."
```

Comment:
Why is it guaranteed that the conflict is always between project and package. E.g. why could it not be between entities of same type?

Reply:
De-duplication with sorted set relies solely on the Comparable interface? (not equals / hashcode?)

Reply:
Interesting point. Could it be a problem that the `compareTo()` implementation is not fully aligned with `equals()`? (Since it may return *true* for objects that are not equal.)

Reply:
TBH I'd need to research this for Kotlin, I only know the Java contracts of the functions...

Reply:
> De-duplication with sorted set relies solely on the Comparable interface? (not equals / hashcode?)

Our `getDuplicates()` relies on `groupBy()` which in turn replies on deduplication of keys in a map, which AFAIK relies on `equals()`, not on `compareTo()`.

> TBH I'd need to research this for Kotlin, I only know the Java contracts of these functions...

There should be no different semantics between Kotlin and Java here...

Reply:
From the docs of `SortedSet`: 

> This is so because the Set interface is defined in terms of the equals operation, but a sorted set performs all element comparisons using its compareTo (or compare) method, so two elements that are deemed equal by this method are, from the standpoint of the sorted set, equal.

https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/SortedSet.html

Reply:
But why are you referring to `SortedSet` at all here? The `map` operation turns the `SortedSet`s into `List`s.

Reply:
Because the initial question was if `projects` and `packages` could each contain duplicate identifiers. But since they are both `SortedSet`s, they can't. Therefore, if both are mapped to one list of identifiers, all duplicates must be between projects and packages.

Reply:
So, if this issue is now clarified, is there anything else missing for an approval?